pcb-json-to-tscircuit-soup-json/parse-pad-name.ts

```diff
@@ -0,0 +1,108 @@
+import { mm } from "@tscircuit/mm"
+
+type PadInfo = {
+  shape: string
+  layer: "A" | "B" | "T"
+  width: number | null
+  height: number | null
+  diameter?: number
+  roundedCornerRadius?: number
+  customParams?: string[]
+}
+
+export function parsePadName(padName: string): PadInfo {
+  const parts = padName.split("_")
+  const [shapeAndLayer, ...rest] = parts
+
+  if (!shapeAndLayer) {
+    throw new Error(`Unable to parse pad shape and layer from ${padName}`)
+  }
+
+  const shapeMatch = shapeAndLayer.match(/([A-Za-z]+)(?:\[([ABT])\])?/)
```

Comment:
do we still need this file? I thought we didn't need to parse the pad names anymore